service/src/main/java/uk/gov/hmcts/reform/fpl/service/cafcass/api/CafcassApiCaseDocumentsConverter.java

```diff
@@ -0,0 +1,257 @@
+                    c2Bundle.getSupportingEvidenceBundle()));
+                c2DocRef.addAll(unwrapElements(c2Bundle.getDraftOrdersBundle()).stream()
+                    .map(DraftOrder::getDocument).toList());
+
+                resultList.addAll(buildCafcassApiCaseDocumentList(C2_APPLICATION_DOCUMENTS, false, c2DocRef));
+            }
+
+            OtherApplicationsBundle otherBundle = additionalApplicationsBundle.getOtherApplicationsBundle();
+            if (isNotEmpty(otherBundle)) {
+                List<DocumentReference> otherDocRef = new ArrayList<>();
+
+                otherDocRef.add(otherBundle.getDocument());
+                otherDocRef.addAll(getAllDocumentsFromSupplements(otherBundle.getSupplementsBundle()));
+                otherDocRef.addAll(getAllDocumentsFromSupportingEvidenceBundles(
+                    otherBundle.getSupportingEvidenceBundle()));
+
+                resultList.addAll(buildCafcassApiCaseDocumentList(C1_APPLICATION_DOCUMENTS, false, otherDocRef));
+            }
+        });
+
+        return resultList;
+    }
+
+    private List<DocumentReference> getAllDocumentsFromSupportingEvidenceBundles(List<Element<SupportingEvidenceBundle>>
+                                                                                     bundles) {
+        return unwrapElements(bundles).stream()
+            .flatMap(supportingEvidenceBundleElement -> Stream.of(
+                supportingEvidenceBundleElement.getDocument(),
+                supportingEvidenceBundleElement.getTranslatedDocument()))
+            .toList();
+    }
+
+    private List<DocumentReference> getAllDocumentsFromSupplements(List<Element<Supplement>> bundles) {
+        return unwrapElements(bundles).stream().map(Supplement::getDocument).toList();
+    }
+
+    private List<CafcassApiCaseDocument> getHearingNotice(CaseData caseData) {
+        return buildCafcassApiCaseDocumentList(NOTICE_OF_ACTING_OR_ISSUE, false,
+            unwrapElements(caseData.getHearingDetails()).stream()
+                .flatMap(hearingBooking -> Stream.of(
+                    hearingBooking.getNoticeOfHearing(),
+                    hearingBooking.getTranslatedNoticeOfHearing())));
+    }
+
+    private List<CafcassApiCaseDocument> getManageDocuments(CaseData caseData) {
+        return Arrays.stream(DocumentType.values())
+            .filter(documentType -> isNotEmpty(documentType.getBaseFieldNameResolver()))
+            .map(documentType -> buildCafcassApiCaseDocumentList(documentType, false,
+                Stream.of(ConfidentialLevel.NON_CONFIDENTIAL, ConfidentialLevel.LA)
+                    .map(confidentialLevel -> manageDocumentService
+                        .toFieldNameToListOfElementMap(caseData, documentType, confidentialLevel).values())
+                    .flatMap(Collection::stream).flatMap(List::stream)
+                    .map(Element::getValue)
+                    //.filter(object -> object instanceof WithDocument)
```

Comment:
same thing here, should we be keeping those filters?

Reply:
think we should let it throws Class cast exception, so that we won't miss any documents